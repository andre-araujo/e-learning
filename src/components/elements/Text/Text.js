import React from 'react';
import { node, string } from 'prop-types';

import {
  StyledText,
} from './Text.styles';

function Text({
  children,
  tag,
  type,
  margin,
  color,
  ...props
}) {
  const Component = StyledText(tag);
  return (
    <Component {...props} type={type} margin={margin} color={color}>
      {children}
    </Component>
  );
}

Text.Title = props => <Text type="title" tag="h1" {...props} />;
Text.Subtitle = props => <Text type="subtitle" tag="h2" {...props} />;
Text.Paragraph = props => <Text type="paragraph" tag="p" {...props} />;
Text.Small = props => <Text type="small" tag="span" {...props} />;

Text.defaultProps = {
  children: null,
  tag: 'span',
  type: 'paragraph',
  color: null,
};

Text.propTypes = {
  children: node,
  tag: string,
  type: string,
  color: string,
};

export default Text;
