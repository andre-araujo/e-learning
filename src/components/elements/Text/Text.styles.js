import styled, { css } from 'styled-components';

export const StyledText = tag => styled[tag]`
  font-family: Roboto;

  ${({ type }) => {
    if (type === 'title') {
      return css`
        font-size: 1.5rem;
        font-weight: bolder;
        color: #333;

        @media (min-width: 768px) {
          font-size: 2rem;
        }
      `;
    }

    if (type === 'subtitle') {
      return css`
        font-weight: bold;
        font-size: 1rem;
        color: #555;

        @media (min-width: 768px) {
          font-size: 1.2rem;
        }
      `;
    }

    if (type === 'paragraph') {
      return css`
        font-size: 0.875rem;
        font-weight: 400;
        color: #000;
    `;
    }

    if (type === 'small') {
      return css`
        font-size: 0.75rem;
        font-weight: 200;
        color: #888;
    `;
    }

    if (type === 'big') {
      return css`
        font-size: 2.5rem;
        font-weight: bolder;
        color: #888;

        @media (min-width: 768px) {
          font-size: 4rem;
        }
    `;
    }

    return null;
  }}

  ${({ margin }) => margin && css`
    margin: ${margin};
  `}

  ${({ color }) => color && css`
    color: ${color};
  `}
`;
