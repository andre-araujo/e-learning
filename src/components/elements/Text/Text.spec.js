import React from 'react';
import { shallow } from 'enzyme';

import Text from './Text';

describe('Text component', () => {
  it('should match snapshot', () => {
    expect(shallow(<Text />)).toMatchSnapshot();
  });
  it('should match snapshot for Big', () => {
    expect(shallow(<Text.Big />)).toMatchSnapshot();
  });
  it('should match snapshot for Title', () => {
    expect(shallow(<Text.Title />)).toMatchSnapshot();
  });
  it('should match snapshot for Subtitle', () => {
    expect(shallow(<Text.Subtitle />)).toMatchSnapshot();
  });
  it('should match snapshot for Paragraph', () => {
    expect(shallow(<Text.Paragraph />)).toMatchSnapshot();
  });
  it('should match snapshot for Small', () => {
    expect(shallow(<Text.Small />)).toMatchSnapshot();
  });
});
