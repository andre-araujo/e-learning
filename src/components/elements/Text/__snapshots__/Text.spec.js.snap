// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should match snapshot 1`] = `
<styled.span
  color={null}
  type="paragraph"
/>
`;

exports[`Text component should match snapshot for Big 1`] = `
<Text
  color={null}
  tag="h1"
  type="big"
/>
`;

exports[`Text component should match snapshot for Paragraph 1`] = `
<Text
  color={null}
  tag="p"
  type="paragraph"
/>
`;

exports[`Text component should match snapshot for Small 1`] = `
<Text
  color={null}
  tag="span"
  type="small"
/>
`;

exports[`Text component should match snapshot for Subtitle 1`] = `
<Text
  color={null}
  tag="h2"
  type="subtitle"
/>
`;

exports[`Text component should match snapshot for Title 1`] = `
<Text
  color={null}
  tag="h1"
  type="title"
/>
`;
